src/edu/csus/ecs/pc2/ui/ResultsComparePane.java

```diff
@@ -0,0 +1,461 @@
+        selectExportDirectoryButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                selectUpdateExportDirectory(pc2ResultsDirectoryTextField);
+            }
+        });
+        selectExportDirectoryButton.setToolTipText("Select export Directory");
+        
+        JPanel summaryResultsPane = new JPanel();
+        summaryResultsPane.setPreferredSize(new Dimension(800, 300));
+        summaryResultsPane.setMinimumSize(new Dimension(300, 300));
+        summaryResultsPane.setBorder(new TitledBorder(null, "Compare Results Summary", TitledBorder.LEADING, TitledBorder.TOP, null, null));
+        centerPane.add(summaryResultsPane);
+        summaryResultsPane.setLayout(new BorderLayout(0, 0));
+        
+        JPanel resCenterPane = new JPanel();
+        summaryResultsPane.add(resCenterPane, BorderLayout.CENTER);
+        resCenterPane.setLayout(new BorderLayout(0, 0));
+        textArea.setFont(new Font("Courier New", Font.PLAIN, 13));
+        
+        
+//        resCenterPane.add(textArea);
+        
+        JScrollPane scrollPane = new JScrollPane(textArea);
+        resCenterPane.add(scrollPane, BorderLayout.CENTER);
+        
+        JPanel resButtonPane = new JPanel();
+        FlowLayout flowLayout = (FlowLayout) resButtonPane.getLayout();
+        flowLayout.setHgap(45);
+        summaryResultsPane.add(resButtonPane, BorderLayout.SOUTH);
+        
+        JButton viewReportButton = new JButton("View Report");
+        viewReportButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                viewCompareResultsFiles();
+            }
+        });
+        resButtonPane.add(viewReportButton);
+        
+        
+        showDetailsCheckbox = new JCheckBox("Include details in comparison");
+        resButtonPane.add(showDetailsCheckbox);
+        showDetailsCheckbox.setToolTipText("Show full detils in comparison");
+        FlowLayout flowLayout_1 = (FlowLayout) southPane.getLayout();
+        flowLayout_1.setHgap(45);
+        
+        southPane.setSize(30,30);
+        add(southPane, BorderLayout.SOUTH);
+
+        JButton exportResultsButton = new JButton("Export Results");
+        southPane.add(exportResultsButton);
+        exportResultsButton.setToolTipText("Export Results files to pc2 results directory");
+
+        JButton compartResultsButton = new JButton(SHOW_COMPARISON_BUTTON_TITLE);
+        southPane.add(compartResultsButton);
+        compartResultsButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                updateSummary();
+            }
+        });
+        compartResultsButton.setToolTipText("Compare pc2 results files to primary CCS results");
+        exportResultsButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                exportpc2ResultsFiles();
```

Comment:
This method call has the potential to last a long time -- in particular, because it can open up a dialog and wait for the user to respond.  During that time the GUI will be frozen (more correctly, the Swing thread will be blocked from processing further Swing events while it waits for the processing of this `ActionEvent` to complete).  I think this needs to be spun off on its own thread -- or at least, invoked via `invokeLater()`.

I'm not sure whether this is a SHOW STOPPER or not -- but it certainly seems close...

Reply:
I will add the invokeLater in case you are correct, but I am not convinced in this case that it is necessary.  In general yes, in this case I have my doubts.